### PR TITLE
More type safety for early muxer negotiation 

### PR DESCRIPTION
**Pull request by Nashatyrev**
Since there are no alternatives at the moment to early negotiate anything beside muxer I refactor the code a bit to add type safety: 
- The list of `StreamMuxer`s is passed to the `SecureChannel` instead of list of Strings 
- The `SecureChannel` is optionally set `NegotiatedStreamMuxer` in the `Session`

### CODE REVIEW

File: libp2p/src/main/kotlin/io/libp2p/core/security/SecureChannel.kt [hunk 1]
```diff
@@ -24,9 +26,11 @@ interface SecureChannel : ProtocolBinding<SecureChannel.Session> {
...
 
-        /** The id of the next protocol, used to select the muxer
-         *
+        /**
+         * Contains muxer if security protocol supports
+         * [Early Multiplexer Negotiation](https://docs.libp2p.io/concepts/multiplex/early-negotiation/)
+         * and the protocol was successfully negotiated. Else contains `null`
          */
-        val nextProto: String
+        val earlyMuxer: NegotiatedStreamMuxer?
```
Comment (ianopolous) [inline libp2p/src/main/kotlin/io/libp2p/core/security/SecureChannel.kt:34]: Is it worth making this an Optional?

**Review by ianopolous — APPROVED** [review]
All looks good, modulo 1 comment. Thanks for cleaning it up.

File: libp2p/src/main/kotlin/io/libp2p/core/security/SecureChannel.kt (hunk 1 above)
Comment (Nashatyrev) [reply libp2p/src/main/kotlin/io/libp2p/core/security/SecureChannel.kt:34]: It's kind of 'optional' in the sense of 'nullable'

Comment (ianopolous) [reply libp2p/src/main/kotlin/io/libp2p/core/security/SecureChannel.kt:34]: Yep, I just meant using the type system to make it obvious.

Comment (ianopolous) [reply libp2p/src/main/kotlin/io/libp2p/core/security/SecureChannel.kt:34]: Maybe this is the Kotlin way to do that?

Comment (Nashatyrev) [reply libp2p/src/main/kotlin/io/libp2p/core/security/SecureChannel.kt:34]: Yeah, I believe its the Kotlin way :) 
I.e. usually a nullable type is used for a reason and should be handled explicitly.
Actually Java `Optional` is equivalent to nullable type in Kotlin. The latter is more convenient of course :)

**Pull request merged**